src/main/java/org/wildfly/security/ssl/Protocol.java

```diff
@@ -29,40 +29,51 @@
     /**
      * The SSL version 2 protocol.
      */
-    SSLv2,
+    SSLv2 ("SSLv2"),
     /**
      * The SSL version 3 protocol.
      */
-    SSLv3,
+    SSLv3 ("SSLv3"),
     /**
      * The TLS version 1.0 protocol.
      */
-    TLSv1,
+    TLSv1 ("TLSv1"),
     /**
      * The TLS version 1.1 protocol.  Note that there are no cipher suites which are specifically defined in this protocol.
      */
-    TLSv1_1,
+    TLSv1_1 ("TLSv1.1"),
     /**
      * The TLS version 1.2 protocol.
      */
-    TLSv1_2,
+    TLSv1_2 ("TLSv1.2"),
     /**
      * The TLS version 1.3 protocol.
      */
-    TLSv1_3,
+    TLSv1_3 ("TLSv1.3"),
     ;
+
     static final int fullSize = values().length;
 
-    static Protocol forName(final String name) {
-        switch (name) {
-            case "SSLv2": return SSLv2;
-            case "SSLv3": return SSLv3;
-            case "TLSv1": return TLSv1;
-            case "TLSv1.1": return TLSv1_1;
-            case "TLSv1.2": return TLSv1_2;
-            case "TLSv1.3": return TLSv1_3;
-            default: return null;
+    public final String name;
+
+    Protocol(String name) {
+        this.name = name;
+    }
+
+    public static Protocol forName(final String name) {
+        for (Protocol protocol : values()) {
+            if (protocol.name.equals(name)) return protocol;
+        }
+        return null;
+    }
+
+    public static String[] names() {
```

Comment:
The usecase is to provide set of allowed values for subsystem - I would replace it with map.keysSet() from map you suggested above, but to prevent map modifications from outside the returning of array sounds better then returning the set enclosed into UnmodifiableSet...